raiden/transfer/channel.py

```diff
@@ -1924,91 +2019,200 @@ def handle_action_set_reveal_timeout(
+        and coop_settle_state.partner_signature_confirmation is not None
+        and coop_settle_state.expiration >= block_number - channel_state.reveal_timeout
+    ):
+        send_coop_settle = ContractSendChannelCoopSettle(
             canonical_identifier=channel_state.canonical_identifier,
-            recipient=channel_state.partner_state.address,
-            recipient_metadata=withdraw_request.sender_metadata,
-            message_identifier=withdraw_request.message_identifier,
-            total_withdraw=withdraw_request.total_withdraw,
-            participant=channel_state.partner_state.address,
-            nonce=channel_state.our_state.nonce,
-            expiration=withdraw_state.expiration,
+            our_total_withdraw=coop_settle_state.total_withdraw_participant,
+            partner_total_withdraw=coop_settle_state.total_withdraw_partner,
+            expiration=coop_settle_state.expiration,
+            signature_our_withdraw=coop_settle_state.partner_signature_confirmation,
+            signature_partner_withdraw=coop_settle_state.partner_signature_request,
+            triggered_by_block_hash=block_hash,
         )
+        return [send_coop_settle]
+    return []
 
-        events: List[Event] = [send_withdraw]
-    else:
-        error_msg = is_valid.as_error_message
+
+def handle_receive_withdraw_request(
+    channel_state: NettingChannelState,
+    withdraw_request: ReceiveWithdrawRequest,
+    block_number: BlockNumber,
+    block_hash: BlockHash,
+    pseudo_random_generator: random.Random,
+) -> TransitionResult[NettingChannelState]:
+    events: List[Event] = list()
+
+    def transition_result_invalid(
+        success_or_error: SuccessOrError,
+    ) -> TransitionResult[NettingChannelState]:
+        error_msg = success_or_error.as_error_message
         assert error_msg, "is_valid_withdraw_request should return error msg if not valid"
         invalid_withdraw_request = EventInvalidReceivedWithdrawRequest(
             attempted_withdraw=withdraw_request.total_withdraw, reason=error_msg
         )
-        events = [invalid_withdraw_request]
+        return TransitionResult(channel_state, [invalid_withdraw_request])
+
+    is_valid = is_valid_withdraw_request(
+        channel_state=channel_state, withdraw_request=withdraw_request
+    )
+    if not is_valid:
+        return transition_result_invalid(is_valid)
+
+    withdraw_state = PendingWithdrawState(
+        total_withdraw=withdraw_request.total_withdraw,
+        nonce=withdraw_request.nonce,
+        expiration=withdraw_request.expiration,
+        recipient_metadata=withdraw_request.sender_metadata,
+    )
+    channel_state.partner_state.withdraws_pending[withdraw_state.total_withdraw] = withdraw_state
+    channel_state.partner_state.nonce = withdraw_request.nonce
+
+    our_initiated_coop_settle = channel_state.our_state.initiated_coop_settle
+
+    if our_initiated_coop_settle is not None or withdraw_request.coop_settle is True:
```

Comment:
We should avoid checking booleans with `is`. Just use plain `or withdraw_request.coop_settle:`

Reply:
Done.